Add Vietnamese language

shitcreek opened the pull request:
Added Vietnamese translation.

My Vietnamese is a bit lacking, and so correction is appreciated.

shitcreek committed:
Add Vietnamese LCD language translation

shitcreek committed:
Add Vietnamese LCD language translation

longle2710 commented on Marlin/src/lcd/language/language_vn.h:
```diff
@@ -0,0 +1,1506 @@
+#endif
+#ifndef MSG_MAIN
+// Main
+  #define MSG_MAIN                            _UxGT("Chánh")
```
Main in Vietnamese should be: "Chính", "Chánh" is an old way to say it

longle2710 commented on Marlin/src/lcd/language/language_vn.h:
```diff
@@ -0,0 +1,1506 @@
+#endif
+#ifndef MSG_EXTRUDE
+// Extrude
+  #define MSG_EXTRUDE                         _UxGT("Ép đùn")
```
Extrude can be roughly translate into:" Đùn nhựa "

shitcreek replied:
I can go either way with this. Just 'Đùn' alone is fine too.

longle2710 commented on Marlin/src/lcd/language/language_vn.h:
```diff
@@ -0,0 +1,1506 @@
+#endif
+#ifndef MSG_LEVEL_BED
+// Level bed
+  #define MSG_LEVEL_BED                       _UxGT("Làm bằng mặt bàn")
```
Bed level can be translate to: " Cân chỉnh bàn in "

shitcreek replied:
I think 'Chỉnh bàn' would be good although since 'chỉnh' means adjust, it is a bit vague. 
Làm bàn bằng (with 'mặt' omitted) could work.

shitcreek replied:
Also, I use 'chỉnh' for 'tune' in other places

longle2710 commented on Marlin/src/lcd/language/language_vn.h:
```diff
@@ -0,0 +1,1506 @@
+#if LCD_WIDTH >= 20
+  #ifndef MSG_INFO_PRINT_COUNT
+// Print Count
+    #define MSG_INFO_PRINT_COUNT              _UxGT("Số In")
```
Print count can be translate to:"Số lần đã in"

shitcreek replied:
I think 'Số In' is fine here.

thinkyhead commented on Marlin/src/lcd/dogm/fontdata/langdata_vi.h:
```diff
@@ -0,0 +1,194 @@
+    FONTDATA_ITEM(61, 235, 235, fontpage_61_235_235), // 'ừ' -- 'ừ'
+    FONTDATA_ITEM(61, 237, 237, fontpage_61_237_237), // 'ử' -- 'ử'
+    FONTDATA_ITEM(61, 239, 239, fontpage_61_239_239), // 'ữ' -- 'ữ'
+    FONTDATA_ITEM(61, 241, 241, fontpage_61_241_241), // 'ự' -- 'ự'
```
@AnHardt — As someone with BDF experience, do you know if we can use this list to figure out what to add to the `marlin-6x12-3.bdf` file? I assume that 61 is the UTF8 page and 241 is the index for the glyph. After that, we're stuck.

I see that the first character above (`ă`) has a fontpage of 2 with index 131. And in the BDF file we have `STARTCHAR abreve \n ENCODING 259`, where 259 happens to be 128 + 131. Coincidence?

I may be able to hunt down a BDF file with the right glyphs, but maybe not 6x12. We'll keep looking!

yhfudev replied:
@thinkyhead The page and value are for u8glib+marlin only. Here's the formula to calculate the unicode value from the page+value: page * 128 + (begin-128). For example 'ử', the page=61, and begin=237 in the header file, so the unicode is 61*128+237-128=7917(0x1EED).

Another way to get the encoding value is to copy the 'ử' to a file and save it with encoding UTF16LE or UTF16BE, and open it with a hex editor.

thinkyhead committed:
Add vi to LCD_LANGUAGE values list

thinkyhead committed:
Add vi to LCD_LANGUAGE values list

Roxy-3D committed:
Remove redundant Frivolous Game Options

shitcreek committed:
Merge branch 'bugfix-2.0.x' of https://github.com/shitcreek/Marlin into bugfix-2.0.x

shitcreek committed:
Update ultralcd_DOGM.h

Fix for FYSET_MINI_12864 display and tested, might have fixed MINIPANEL as well, will need to be tested. Issue was fixed with replacing 'Either' case with seperate 'Enabled' cases
